services/order.py

```diff
@@ -0,0 +1,33 @@
+from django.contrib.auth import get_user_model
+from django.db import transaction
+
+from db.models import Order, Ticket, MovieSession
+
+
+def create_order(
+        tickets: list[dict],
+        username: str,
+        date: str = None
+) -> None:
+    with transaction.atomic():
+        user = get_user_model().objects.get(username=username)
+        order = Order.objects.create(user=user)
+        if date:
+            order.created_at = date
+            order.save()
+        for ticket_data in tickets:
+            Ticket.objects.create(
+                order=order,
+                movie_session=MovieSession.objects.get(
+                    id=ticket_data["movie_session"]
+                ),
+                row=ticket_data["row"],
+                seat=ticket_data["seat"]
+            )

```

Comment:
Use  `@transaction.atomic ` decorator to wrap the entire function. Using the decorator makes the code cleaner and more concise while still preserving the atomicity of the transaction